Storage/SQL/SQLiteBookmarksBase.swift

```diff
@@ -55,15 +55,18 @@ open class SQLiteBookmarks: BookmarksModelFactorySource, KeywordSearchSource {
         let sql = "SELECT bmkUri FROM view_bookmarksBuffer_on_mirror WHERE keyword = ? LIMIT 1"
         let args: Args = [keyword]
 
-        return self.db.runQueryConcurrently(sql, args: args, factory: { $0["bmkUri"] as! String })
+        let deferred = DeferredDBOperation<Maybe<String>>()
+
+        self.db.runQueryConcurrently(sql, args: args, factory: { $0["bmkUri"] as? String })
             >>== { cursor in
-                if cursor.status == .success {
-                    if let str = cursor[0] {
-                        return deferMaybe(str)
-                    }
-                }
+            guard cursor.status == .success, let str = cursor[0] as? String else {
+                deferred.fill(Maybe(failure: NoSuchSearchKeywordError(keyword: keyword)))
+                return
+            }
 
-                return deferMaybe(NoSuchSearchKeywordError(keyword: keyword))
+            deferred.fill(Maybe(success: str))
         }
+
+        return deferred
```

Comment:
I had to make these changes because I just noticed that a change I made yesterday to make this query cancellable wasn't working because we were losing the `Cancellable` interface through the `>>==`.